site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2406,250 +2412,465 @@ public function deleteGradeable($g_id) {
+
+        // Setup the mark with its new id
+        $mark->setIdFromDatabase($this->course_db->getLastInsertId());
+    }
+
+    /**
+     * Updates a mark in the database
+     * @param Mark $mark The mark to update
+     */
+    private function updateMark(Mark $mark) {
+        $params = [
+            $mark->getComponent()->getId(),
+            $mark->getPoints(),
+            $mark->getTitle(),
+            $mark->getOrder(),
+            $this->course_db->convertBoolean($mark->isPublish()),
+            $mark->getId()
+        ];
+        $this->course_db->query("
+            UPDATE gradeable_component_mark SET 
+              gc_id=?,
+              gcm_points=?,
+              gcm_note=?,
+              gcm_order=?,
+              gcm_publish=?
+            WHERE gcm_id=?", $params);
+    }
+
+    /**
+     * Deletes an array of marks from the database and any
+     *  data associated with them
+     * @param Mark[] $marks An array of marks to delete
+     */
+    public function deleteMarks(array $marks) {
+        if (count($marks) === 0) {
+            return;
+        }
+        // We only need the ids
+        $mark_ids = array_map(function (Mark $mark) {
+            return $mark->getId();
+        }, $marks);
+        $place_holders = implode(',', array_fill(0, count($marks), '?'));
+
+        $this->course_db->query("DELETE FROM gradeable_component_mark_data WHERE gcm_id IN ($place_holders)", $mark_ids);
+        $this->course_db->query("DELETE FROM gradeable_component_mark WHERE gcm_id IN ($place_holders)", $mark_ids);
+    }
+
+    /**
+     * Creates a new Component in the database
+     * @param Component $component The component to insert
+     */
+    private function createComponent(Component $component) {
+        $params = [
+            $component->getGradeable()->getId(),
+            $component->getTitle(),
+            $component->getTaComment(),
+            $component->getStudentComment(),
+            $component->getLowerClamp(),
+            $component->getDefault(),
+            $component->getMaxValue(),
+            $component->getUpperClamp(),
+            $this->course_db->convertBoolean($component->isText()),
+            $component->getOrder(),
+            $this->course_db->convertBoolean($component->isPeer()),
+            $component->getPage()
+        ];
+        $this->course_db->query("
+            INSERT INTO gradeable_component(
+              g_id, 
+              gc_title, 
+              gc_ta_comment, 
+              gc_student_comment, 
+              gc_lower_clamp, 
+              gc_default, 
+              gc_max_value, 
+              gc_upper_clamp,
+              gc_is_text, 
+              gc_order, 
+              gc_is_peer, 
+              gc_page)
+            VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)", $params);
+
+        // Setup the component with its new id
+        $component->setIdFromDatabase($this->course_db->getLastInsertId());
```

Comment:
Needs an `intval()` around `$this->course_db->getLastInsertId()` which apparently returns a string

Reply:
Same for `createMark`

Reply:
Ids get parsed as strings in all places inside the model classes now (only some were previously)